src/main/java/software/amazon/smithy/lsp/SmithyTextDocumentService.java

```diff
@@ -477,7 +476,25 @@ private Optional<Shape> getTargetShape(Shape initialShape, String token, Model m
                 .findFirst();
     }
 
-    private String getSerializedShape(Shape shape, Model model) {
+    private String getHoverContentsForShape(Shape shape, Model model) {
+        try {
+            String serializedShape = serializeShape(shape, model);
+            return "```smithy\n" + serializedShape + "\n```";
+        } catch (Exception e) {
+            List<ValidationEvent> validationEvents = getValidationEventsForShape(shape);
+            StringBuilder contents = new StringBuilder();
+            contents.append("Can't display shape ").append("`").append(shape.getId().toString()).append("`:");
+            for (ValidationEvent event : validationEvents) {
+                contents.append(System.lineSeparator()).append(event.getMessage());
+            }
+            if (validationEvents.isEmpty()) {
+                contents.append(System.lineSeparator()).append(e);
+            }
+            return contents.toString();
+        }
+    }
```

Comment:
Not sure if this is the best way to go, but my thinking was we only wanted to display the error content if absolutely necessary. There may be a better way to do this that I can't think of at the moment.